src/messaging/messaging.ts

```diff
@@ -245,15 +301,26 @@ export class Messaging implements MessagingInterface {
       throw new FirebaseMessagingError(
         MessagingClientErrorCode.INVALID_ARGUMENT, 'dryRun must be a boolean');
     }
-    return this.getUrlPath()
-      .then((urlPath) => {
-        const request: { message: Message; validate_only?: boolean } = { message: copy };
-        if (dryRun) {
-          request.validate_only = true;
-        }
-        return this.messagingRequestHandler.invokeRequestHandler(FCM_SEND_HOST, urlPath, request);
+
+    const clientMessage = this.convertToClientMessage(copy);
+
+    const client = new FcmServiceClient({
+      credentials: this.getAppKey(),
+      fallback: 'rest'
+    });
+
+    return client.getProjectId()
+      .then((projectId) => {
+        const parent = `projects/${projectId}`;
+
+        const request: ClientSendRequest = {
+          parent: parent,
+          message: clientMessage,
+          validateOnly: dryRun
+        };
+        return client.sendMessage(request);
       })
-      .then((response) => {
+      .then(([response]) => {
```

Comment:
`response` has type `IMessage` with this implementation. We should be able to remove the following `any` cast.

Reply:
Resolved!